react/features/mobile/watchos/middleware.js

```diff
@@ -0,0 +1,203 @@
+import { Platform } from 'react-native';
+import * as watch from 'react-native-watch-connectivity';
+
+import { appNavigate } from '../../app';
+
+import { APP_WILL_MOUNT } from '../../base/app';
+import { CONFERENCE_JOINED } from '../../base/conference';
+import { getInviteURL, isInviteURLReady } from '../../base/connection';
+import { setAudioMuted } from '../../base/media';
+import {
+    MiddlewareRegistry,
+    StateListenerRegistry,
+    toState
+} from '../../base/redux';
+import { toURLString } from '../../base/util';
+
+import { setConferenceTimestamp, setConferenceURL, setMicMuted, setRecentUrls } from './actions';
+
+const logger = require('jitsi-meet-logger').getLogger(__filename);
+
+const watchOSEnabled = Platform.OS === 'ios';
+
+// Handles the recent URLs state sent to the watch
+watchOSEnabled && StateListenerRegistry.register(
+    /* selector */ state => state['features/recent-list'],
+    /* listener */ (recentListState, { dispatch, getState }) => {
+        dispatch(setRecentUrls(recentListState));
+        _updateApplicationContext(getState);
+    });
+
+// Handles the mic muted state sent to the watch
+watchOSEnabled && StateListenerRegistry.register(
+    /* selector */ state => _isAudioMuted(state),
+    /* listener */ (isAudioMuted, { dispatch, getState }) => {
+        dispatch(setMicMuted(isAudioMuted));
+        _updateApplicationContext(getState);
+    });
+
+// Handles the conference URL state sent to the watch
+watchOSEnabled && StateListenerRegistry.register(
+    /* selector */ state => _getCurrentConferenceUrl(state),
+    /* listener */ (currentUrl, { dispatch, getState }) => {
+        dispatch(setConferenceURL(currentUrl));
+        _updateApplicationContext(getState);
+    });
+
+/**
+ * Middleware that captures conference actions.
+ *
+ * @param {Store} store - The redux store.
+ * @returns {Function}
+ */
+watchOSEnabled && MiddlewareRegistry.register(store => next => action => {
+    switch (action.type) {
+    case APP_WILL_MOUNT:
+        _appWillMount(store);
+        break;
+    case CONFERENCE_JOINED:
+        store.dispatch(setConferenceTimestamp(new Date().getTime()));
+        _updateApplicationContext(store.getState());
+        break;
+    }
+
+    return next(action);
+});
+
+/**
+ * Registers listeners to the react-native-watch-connectivity lib.
+ *
+ * @param {Store} store - The redux store.
+ * @private
+ * @returns {void}
+ */
+function _appWillMount({ dispatch, getState }) {
+
+    watch.subscribeToWatchState((error, watchState) => {
```

Comment:
before this, maybe check if the app is installed with this new api? https://github.com/mtford90/react-native-watch-connectivity/blob/master/Libraries/RNWatch/RNWatch.ios.js#L246